public/app/features/explore/Explore.tsx

```diff
@@ -324,6 +325,12 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
     const showRichHistory = openDrawer === ExploreDrawer.RichHistory;
     const showQueryInspector = openDrawer === ExploreDrawer.QueryInspector;
     const showLogsVolume = !!logsVolumeDataProvider;
+    // Only show node graph if X-Ray datasource or node graph is enabled
+    const nodeGraphEnabled =
```

Comment:
Shouldn't this be checked at data source level? Even when graph is disabled, frames are processed and passed as `nodeGraphFrames` down to the component and not used anywhere.

Reply:
You're right, I should also check at the data source level since generating the node graph can be relatively expensive. I'll still need to check here though to hide the container. 

Or alternatively I could add a check in the NodeGraphContainer that says 'if no dataframes provided, render null' which would hide it and remove the need for the check below. What do you think?